src/msg/Messenger.h

```diff
@@ -140,6 +141,14 @@ class Messenger {
   virtual ~Messenger() {}
 
   /**
+   * Proplist notation for passing general parameters to Messenger::create()
+   * to support per-Messenger instance creation parameters
+   */
+  typedef boost::variant< int32_t, std::string > PropElt;
```

Comment:
If we're going to bake this data structure into the interface, it would be nice to support more than int and string. Would `boost::any` be crazy here?